web/docs/auth/auth-hooks.md

```diff
@@ -567,3 +722,132 @@ The hook receives an object as **input** with the following properties:
   The [Express request object](https://expressjs.com/en/api.html#req) from which you can access the request headers, cookies, etc.
 
 This hook's return value must be an object that looks like this: `{ url: URL }`. Wasp uses the URL to redirect the user to the OAuth provider.
+
+### The `onBeforeLogin` hook
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```js title="src/auth/hooks.js"
+export const onBeforeLogin = async ({
+  providerId,
+  prisma,
+  req,
+}) => {
+  // Hook code goes here
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title="src/auth/hooks.ts"
+import type { OnBeforeLoginHook } from 'wasp/server/auth' 
+
+export const onBeforeLogin: OnBeforeLoginHook = async ({
+  providerId,
+  prisma,
+  req,
+}) => {
+  // Hook code goes here
+}
+```
+</TabItem>
+</Tabs>
+
+The hook receives an object as **input** with the following properties:
+- [`providerId: ProviderId`](#providerid-fields)
+
+- `prisma: PrismaClient`
+
+  The Prisma client instance which you can use to query your database.
```

Comment:
Should we somehow address the duplication for all these hook API reference bits?

I hate dealing with duplication in docs. It's never obvious to me what's the right thing to do.

Reply:
I've worried about that myself but didn't pull the trigger on extracting them.

Now I've extracted them into `Common hook input` and link from each hook.